araig_interpreters/src/location_interpreters/vicon_interpreter.cpp

```diff
@@ -0,0 +1,135 @@
+#include <ros/ros.h>
+#include <tf/transform_broadcaster.h>
+#include <tf/transform_listener.h>
+#include <geometry_msgs/TransformStamped.h>
+#include <geometry_msgs/PoseStamped.h>
+#include <std_msgs/String.h>
+#include <ros/master.h>
+#include <vector>
+#include <map>
+#include <string>
+#include <boost/algorithm/string.hpp>
+#include <boost/bind.hpp>
+
+class vicon_interpreter
+{
+private:
+    ros::NodeHandle nh;
```

Comment:
You declare two node handles here - you should only need one. Also, both of these two are the same, even though you name your second as priv, which you never use by the way.   

Node handles are special in ros. Your class should not create one itself, but receive it from main. Typically, you let the constructor accept a node handle and you just use that. The main creates the NH and passes it to class.